include/Zydis/Disassemble.h

```diff
@@ -0,0 +1,130 @@
+     * also interested in implicit operands (for example the registers implicitly accessed by
+     * `pushad`). Unused entries are zeroed.
+     */
+    ZydisDecodedOperand operands[ZYDIS_MAX_OPERAND_COUNT];
+    /**
+     * The textual, human-readable representation of the instruction.
+     *
+     * Guaranteed to be zero-terminated.
+     */
+    char text[128];
+} ZydisDisassembledInstruction;
+
+/* ============================================================================================== */
+/* Exported functions                                                                             */
+/* ============================================================================================== */
+
+/**
+ * Disassemble an instruction and format it to human-readable text in a single step (Intel syntax).
+ *
+ * @param machine_mode      The machine mode to assume when disassembling. When in doubt, pass
+ *                          `ZYDIS_MACHINE_MODE_LONG_64` for what is typically referred to as
+ *                          "64-bit mode" or `ZYDIS_MACHINE_MODE_LEGACY_32` for "32-bit mode".
+ * @param runtime_address   The program counter (`eip` / `rip`) to assume when formatting the
+ *                          instruction. Many instructions behave differently depending on the
+ *                          address they are located at.
+ * @param buffer            A pointer to the raw instruction bytes that you wish to decode.
+ * @param length            The length of the input buffer. Note that this can be bigger than the
+ *                          actual size of the instruction -- you don't have to know the size up
+ *                          front. This length is merely used to prevent Zydis from doing
+ *                          out-of-bounds reads on your buffer.
+ * @param instruction       A pointer to receive the decoded instruction information. Can be
+ *                          uninitialized and reused on later calls.
+ *
+ * This is a convenience function intended as a quick path for getting started with using Zydis.
+ * It internally calls a range of other more advanced functions to obtain all commonly needed
+ * information about the instruction. It is likely that you won't need most of this information in
+ * practice, so it is advisable to instead call these more advanced functions directly if you're
+ * concerned about performance.
+ *
+ * This function essentially combines the following more advanced functions into a single call:
+ *
+ *   - `ZydisDecoderInit`
+ *   - `ZydisDecoderDecodeInstruction`
+ *   - `ZydisDecoderDecodeOperands`
+ *   - `ZydisFormatterInit`
+ *   - `ZydisFormatterFormatInstruction`
+ *
+ * @return  A zyan status code.
+ */
+ZYDIS_EXPORT ZyanStatus ZydisDisassembleIntel(ZydisMachineMode machine_mode,
+    ZyanUSize runtime_address, const void* buffer, ZyanUSize length,
+    ZydisDisassembledInstruction *instruction);
```

Comment:
I went with one function for the two major formatter flavors here to keep the argument count low. I don't have a strong take on this and am open towards instead making this a regular argument.